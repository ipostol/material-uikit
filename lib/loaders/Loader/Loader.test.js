'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _enzyme = require('enzyme');

var _Loader = require('./Loader');

var _Loader2 = _interopRequireDefault(_Loader);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

describe('Loader component', function () {

  it('should render correctly loading', function () {

    var props = {
      status: 'loading',
      message: 'message'
    };

    var wrapper = (0, _enzyme.shallow)(_react2.default.createElement(_Loader2.default, props));

    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly success', function () {

    var props = {
      status: 'success',
      message: 'message',
      right: true
    };

    var wrapper = (0, _enzyme.shallow)(_react2.default.createElement(_Loader2.default, props));

    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly error', function () {

    var props = {
      status: 'error'
    };

    var wrapper = (0, _enzyme.shallow)(_react2.default.createElement(_Loader2.default, props));

    expect(wrapper).toMatchSnapshot();
  });
});