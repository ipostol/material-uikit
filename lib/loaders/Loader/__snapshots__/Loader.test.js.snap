// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should render correctly error 1`] = `
<div
  className="general"
>
  <div
    className="loader error"
  >
    <Error />
  </div>
</div>
`;

exports[`Loader component should render correctly loading 1`] = `
<div
  className="general"
>
  message
  <div
    className="loader rotate marginLeft"
  />
</div>
`;

exports[`Loader component should render correctly success 1`] = `
<div
  className="general messageRight"
>
  message
  <div
    className="loader marginRight"
  >
    <Success />
  </div>
</div>
`;
